### PR TITLE
Introduce the concept of a "checking" implementation

### PR DESCRIPTION
**Pull request by cabo**

### CODE REVIEW

File: draft-ietf-jsonpath-iregexp.md [hunk 1]
```diff
@@ -124,14 +124,32 @@ In particular, full Unicode support is REQUIRED; the implementation
 MUST NOT limit itself to 7- or 8-bit character sets such as ASCII and
 MUST support the Unicode character property set in character classes.
 
+## Checking I-Regexp Implementations {#checking}
```
Comment (timbray) [inline draft-ietf-jsonpath-iregexp.md]: "I-Regexp" superfluous, suggest "Checking Implementations"  for title

File: draft-ietf-jsonpath-iregexp.md [hunk 2]
```diff
@@ -124,14 +124,32 @@ In particular, full Unicode support is REQUIRED; the implementation
...
+
+A *checking* I-Regexp implementation only accepts the I-Regexp syntax
```
Comment (timbray) [inline draft-ietf-jsonpath-iregexp.md]: This is OK, but I think the real goal is to be informed of potential problems.  You might choose to go ahead with a nonconforming regexp anyhow. So another definition could be "A checking I-Regexp implementation is one that has the capability of checking a submitted regexp for compliance with this specification and reporting any problems."

**Review by timbray — CHANGES_REQUESTED** [review]

File: draft-ietf-jsonpath-iregexp.md (hunk 2 above)
Comment (cabo) [reply draft-ietf-jsonpath-iregexp.md]: OK, this exposes the fact that even  checking implementation might not be configured to check, if we define it along its capabilities; I'd like to reserve the term for one that actually does.
(And I'm not so hot about "submitted".)

➔ how about:

A *checking* I-Regexp implementation is one that checks a supplied regexp for compliance with this specification and reports any problems.

Comment (timbray) [reply draft-ietf-jsonpath-iregexp.md]: Works for me.

**Review by timbray — APPROVED** [review]

**Pull request merged**